hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/merkle/MerkleUniqueToken.java

```diff
@@ -247,7 +251,9 @@ public void setNext(final NftNumPair next) {
     }
 
     public byte[] getMetadata() {
-        return metadata;
+        return Optional.ofNullable(metadata)
+                .map(byte[]::clone)
+                .orElse(null);
```

Comment:
Using an `Optional` with the use of `.map().orElse()` here creates unnecessary memory allocations and potentially creates a performance issue here by putting unnecessary strain on the garbage collector.

```suggestion
        return (metadata != null) ? metadata.clone() : null;
```